planner/core/rule_aggregation_elimination.go

```diff
@@ -69,6 +69,39 @@ func (a *aggregationEliminateChecker) tryToEliminateAggregation(agg *LogicalAggr
 	return nil
 }
 
+func (a *aggregationEliminateChecker) tryToEliminateDistinct(agg *LogicalAggregation) {
+	if len(agg.GetGroupByCols()) != 0 {
```

Comment:
I think so. But I am not sure whether there are any special cases.

Reply:
If this check is removed, `TestDAGPlanBuilderAgg` will fail. For the sql `select sum(distinct a), avg(b + c) from t group by d`, it seems that this distinct can't be eliminate.

Reply:
```
[2021-04-07T08:29:59.134Z] FAIL: physical_plan_test.go:338: testPlanSuite.TestDAGPlanBuilderAgg
[2021-04-07T08:29:59.134Z] 
[2021-04-07T08:29:59.134Z] physical_plan_test.go:374:
[2021-04-07T08:29:59.134Z]     c.Assert(core.ToString(p), Equals, output[i].Best, Commentf("for %s", tt))
[2021-04-07T08:29:59.134Z] ... obtained string = "TableReader(Table(t)->HashAgg)->HashAgg"
[2021-04-07T08:29:59.134Z] ... expected string = "TableReader(Table(t))->Projection->HashAgg"
[2021-04-07T08:29:59.134Z] ... for select sum(distinct a), avg(b + c) from t group by d
```
I think we should update the test results for this case, since we can remove the `distinct` now, the aggregation can be pushed down to coprocessor.

Reply:
done